pkg/validations/cluster.go

```diff
@@ -215,3 +216,19 @@ func ValidateK8s128Support(clusterSpec *cluster.Spec) error {
 	}
 	return nil
 }
+
+// ValidateTemplateNameWithKubernetesVersion check if the template name in the machineconfig field includes kubernetes version.
+func ValidateTemplateNameWithKubernetesVersion(spec cluster.Spec) error {
+	replacer := strings.NewReplacer("-", "", ".", "")
+	for _, machineconfig := range spec.CloudStackMachineConfigs {
+		// Replace 1.23, 1-23 to 123 in the template name string.
+		templateName := replacer.Replace(machineconfig.Spec.Template.Name)
+		// Replace 1-23 to 123 in the kubernetesversion string.
+		kubernetesVersion := replacer.Replace(string(spec.Cluster.Spec.KubernetesVersion))
+		// Check if the template name includes kubernetes version.
+		if !strings.Contains(templateName, kubernetesVersion) {
+			return fmt.Errorf("invalid template: cluster kubernetes version is %s but template for machineconfig %s is %s", string(spec.Cluster.Spec.KubernetesVersion), machineconfig.Name, machineconfig.Spec.Template.Name)
```

Comment:
can we add something here about why the template is invalid? like "template should include k8s version blbala"